skywalking-collector/skywalking-collector-worker/src/main/java/com/a/eye/skywalking/collector/worker/storage/Window.java

```diff
@@ -0,0 +1,50 @@
+package com.a.eye.skywalking.collector.worker.storage;
+
+import java.util.HashMap;
+
+/**
+ * @author pengys5
+ */
+public abstract class Window<T extends Data> {
+
+    private Pointer current;
```

Comment:
Why need `Pointer`? You only need a filed, type of `WindowData`. And you can use `==` to judge. Better performance, Easier codes.